Expand MV doc around grace period (#20665)
trinodb/trino: docs/src/main/sphinx/connector/iceberg.md

snapshot-ids of all Iceberg tables that are part of the materialized view's
query in the materialized view metadata. When the materialized view is queried,
the snapshot-ids are used to check if the data in the storage table is up to
date.

The grace period of a materialized view defaults to infinity. After each refresh
with this infinite grace period, the materialized view then behaves like a
static table with the data from the time of the refresh.

> If the data is outdated and the grace period of the materialized view is

Reviewer: ```suggestion
If the data is outdated and the grace period of the materialized view has
```

snapshot-ids of all Iceberg tables that are part of the materialized view's
query in the materialized view metadata. When the materialized view is queried,
the snapshot-ids are used to check if the data in the storage table is up to
date.

The grace period of a materialized view defaults to infinity, and after each

> refresh the materialized view therefore behaves like a static table.

Reviewer: ```suggestion
refresh with an infinite grace period, the materialized view then
behaves like a static table.
```
Might also be worth explicitly calling out

If the data is outdated and the grace period of the materialized view is
elapsed, the materialized view behaves like a normal view, and the data is
queried directly from the base tables.

Detecting outdated data is possible only when the materialized view uses Iceberg
tables only, or when it uses a mix of Iceberg and non-Iceberg tables but some
Iceberg tables are outdated. When the materialized view is based on non-Iceberg
tables, querying it can return outdated data, since the connector has no

> information whether the underlying non-Iceberg tables have changed.

Reviewer 2: When there are non-Iceberg tables, view is considered stale once GRACE elapses.

Author: Tried to pull this in with more rewording.

snapshot-ids of all Iceberg tables that are part of the materialized view's
query in the materialized view metadata. When the materialized view is queried,
the snapshot-ids are used to check if the data in the storage table is up to
date.

Materialized views that use non-Iceberg tables in the query show the [default

> behavior around grace period](mv-grace-period). If all tables are Iceberg

Reviewer: ```suggestion
behavior around grace periods](mv-grace-period). If all tables are Iceberg
```
